test/new-e2e/tests/agent-shared-components/config-refresh/non_core_agents_sync_test.go

```diff
@@ -145,28 +150,23 @@ func createFile(t *testing.T, host *components.RemoteHost, filepath string, opti
 }
 
 // assertAgentsUseKey checks that all agents are using the given key.
-func assertAgentsUseKey(t *testing.T, host *components.RemoteHost, authtoken, key string) {
-	t.Helper()
-
+func assertAgentsUseKey(t assert.TestingT, host *components.RemoteHost, authtoken, key string) {
```

Comment:
You can keep the `t.Helper()` in place so that we can get better stack traces on failures. What do you think?

```
	if h, ok := t.(tHelper); ok {
		h.Helper()
	}
```

Reply:
Oh that's an interesting pattern, I like it !